src/shell/mod.rs

```diff
@@ -1690,42 +1690,26 @@ impl Shell {
         }
     }
 
-    pub fn move_current_window(
-        state: &mut State,
-        seat: &Seat<State>,
+    pub fn move_window_to_workspace<'a>(
```

Comment:
So https://github.com/pop-os/cosmic-comp/pull/239/files#diff-4f1e1533acb2e2645a75fc56f2249115a8c1029a541937726ef8e6449b3b0cecR1710 introduces a `move_window` method that takes a `CosmicMapped` and two `WorkspaceHandles` instead of doing this weird index+output dance, that really only was convenient for the shortcuts.

Looks like the wayland handler can easily make use of that and get rid of a couple of lines used to determining the indices. I can handle the rebase, if you want?

Reply:
Using `WorkspaceHandle` is cleaner, and should be easy to update this PR to. I can update this easily enough when https://github.com/pop-os/cosmic-comp/pull/239 is merged.

Reply:
It is merged, so this can be rebased.